### PR TITLE
Release/1.0.0

### PR DESCRIPTION
**Pull request by cs-yash-patodkar**
Initial Commit

### CODE REVIEW

File: widget/view.controller.js [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    $scope.configForMarkdown = {
+      initialEditType: 'markdown', //wysiwyg
+      previewStyle: 'tab',
+      height: '420px',
+      usageStatistics: false,
+      hideModeSwitch: true,
+      toolbarItems: ['']
+    };
+    //Initial message for conversation 
+    $scope.conversationMessages = [
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:56]: Add these $scope.conversationMessages and $scope.playbookSuggestionMessages in single object so we can change and access them easily.

File: widget/view.controller.js [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+          generatePlaybookDescription(userInput);
+          // generateBotResponse($scope.playbookSuggestionMessages);
+        }
+      }
+    };
+
+    // Function to generate a bot response 
+    function generateBotResponse(userMessage) {
+      $scope.receivingResponse = true;
+      var parametersForPlaybook = {
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:112]: parametersForPlaybook is used multiple time. create a parameterised function to return object.

File: widget/view.controller.js [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        ],
+        "__selectFields": [
+          "id",
+          "name"
+        ]
+      };
+
+      // add a tag to the playbook, get the playbook instead of hardcoding the playbook iri
+      getAllPlaybooks(queryObjectPlaybook).then(function (playbook) {
+        var queryUrl = '/api/triggers/1/notrigger/' + playbook['hydra:member'][0]['uuid'] + '?force_debug=true';
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:256]: instead of whole playbook object only send the UUID

File: widget/view.controller.js [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+        },
+        "useMockOutput": false,
+        "globalMock": false
+      }
+      //trigger playbook and get response when execution is completed
+      getDataFromPlaybook($scope.playbookTags.conversation, parametersForPlaybook).then(function (data) {
+        // $scope.showDescription = true;
+        if (data.result) {
+          if (data.result.query_result) {
+            $scope.receivingResponse = false;
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:128]: Add finally block to set $scope.receivingResponse = false. currently it only set to false on success response on error it is not set.

File: widget/view.controller.js [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+          "useMockOutput": false,
+          "globalMock": false
+        }
+        
+        //trigger playbook and get response when execution is completed
+        getDataFromPlaybook($scope.playbookTags.pBDesignerSteps, parametersForPlaybook).then(function (data) {
+          if (data.result && data.result.data) {
+            $scope.receivingResponse = false;
+            $rootScope.$broadcast("designer:addPlaybookElements", data.result.data);
+            $scope.close();
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:164]: create finally block and add common code in it like close() and $scope.receivingResponse = false. these are common code written multiple time

File: widget/view.controller.js [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+            "data": {
+              "records": [],
+              "task_to_automate": inputText
+            }
+          },
+          "useMockOutput": false,
+          "globalMock": false
+        }
+        getDataFromPlaybook($scope.playbookTags.pBDesignerDescription, parametersForPlaybook).then(function (data) {
+          $scope.showDescription = true;
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:201]: $scope.showDescription = true; is not set to false again. Please check is this needed here.

File: widget/view.controller.js (hunk 4 above)
Comment (cs-yash-patodkar) [reply widget/view.controller.js]: It is just a flag

File: widget/view.controller.js [hunk 7]
```diff
@@ -109,73 +122,65 @@
...
-        },
-        "useMockOutput": false,
-        "globalMock": false
-      }
+      var parametersForPlaybook = returnParameters();
+      parametersForPlaybook.request.data["conversation"] = userMessage;
+      // {
+      //   "input": {},
+      //   "request": {
+      //     "data": {
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js]: Removed Comment code

File: widget/view.controller.js [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+      toolbarItems: ['']
+    };
+
+
+    // //Initial message for conversation 
+    // $scope.conversationMessages = [
+    //   {
+    //     text: "Hi there! How can I help you today? Ask me questions like ‘how can I remove all HTML tags from a string’ OR ‘give some suggestions for responding to Ransomware alerts’ or ‘log samples for Fortigate firewall’...",
+    //     type: "bot"
+    //   }
```
Comment (cs-ajahar-inamdar) [inline widget/view.controller.js:63]: Please remove the commented code in next commit

**Review by cs-ajahar-inamdar — APPROVED** [review]
Looks good

**Pull request closed without merging**